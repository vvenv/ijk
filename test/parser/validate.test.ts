import { expect, test } from 'vitest';
import { parse } from './__helper';

test('tag mismatch', () => {
  expect(parse('{{ for i in j }}{{ endif }}')).toMatchSnapshot();
  expect(parse('{{ for i in j }}1{{ elif }}2{{ endfor }}')).toMatchSnapshot();
});

test('no tag', () => {
  expect(parse('{{}}')).toMatchSnapshot();
});
