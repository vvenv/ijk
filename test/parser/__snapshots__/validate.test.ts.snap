// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`no tag 1`] = `""use strict";let s = "";s+="{{}}";return s;"`;

exports[`tag mismatch 1`] = `""use strict";let s = "";return s;"`;

exports[`tag mismatch 2`] = `""use strict";let s = "";const k_1_0=Object.keys(c.j);const l_1_0=k_1_0.length;for (let i_1_0=0; i_1_0<l_1_0; i_1_0++) {const c_i_1_0={...c,i: c.j[k_1_0[i_1_0]],loop: {index: i_1_0,first: i_1_0 === 0,last: i_1_0 === l_1_0,length: l_1_0}};s+="1";s+=e(c_i_1_0.elif);s+="2";}return s;"`;
